osdep/timer.c

```diff
@@ -49,10 +46,7 @@ void mp_time_init(void)
 
 int64_t mp_time_ns(void)
 {
-    uint64_t r = mp_raw_time_ns() - raw_time_offset;
-    if (r < MP_START_TIME)
-        r = MP_START_TIME;
-    return r;
+    return mp_raw_time_ns() - raw_time_offset;
```

Comment:
My comment yesterday about clocks where not fully correct, because we in fact were ensuring that `mp_time_ns` does not return 0. Which in fact can happen, because we subtract `raw_time_offset` which is starting point.

Ok, but I still don't see a reason why `mp_time_ns` shouldn't return 0.

> offset to avoid confusing relative/absolute times.

So we break absolute time to avoid confusion. And who is the confused one?

> Also,we rule that the timer never returns 0 (so default-initialized time values will be always in the past).

Again, I don't see any gain from that. Even if we were do `0 - 0` timeout somewhere, it is still zero.

That's said I approve change to remove this. I'm just talking at loud, because maybe someone knows any reason to not do that.

Also in fact now the timeout is 10ns it wouldn't even matter in any case.

Reply:
> Ok, but I still don't see a reason why `mp_time_ns` shouldn't return 0.

I've mostly copied that "requirement" from the existing comment about ensuring not returning `0`. Should I drop it?

Reply:
No, everything is fine, I was commenting on the `MP_START_TIME` removal and if it is fine to do so, but I don't see any issue with that.